### PR TITLE
Add CD for spark Job

### PR DESCRIPTION
**Pull request by JoelClemence**
* Upload Spark Job to S3 using Terraform
* Deployment depends on shadowJar of spark-job
* Lambda references latest upload

### CODE REVIEW

File: deployment/build.gradle.kts [hunk 1]
```diff
@@ -18,13 +18,20 @@ configure<SpotlessExtension> {
 
 configure<TerraformExtension> {
     val aisBatchLambdaShadowJarTask = tasks.getByPath(":ais-batch-lambda:shadowJar")
+    val sparkJobShadowJarTask = tasks.getByPath(":spark-job:shadowJar")
 
     dependsOn(
             aisBatchLambdaShadowJarTask
```
Comment (ciaranevans) [inline deployment/build.gradle.kts]: Looks like weird indentation here.

File: deployment/src/main/terraform/main.tf [hunk 2]
```diff
@@ -12,11 +12,13 @@ terraform {
...
-  ais_bucket      = "${data.external.secrets.result["ais_bucket"]}"
-  heatmap_bucket  = "${data.external.secrets.result["heatmap_bucket"]}"
-  emr_logs_bucket = "${data.external.secrets.result["emr_logs_bucket"]}"
+  source             = "./modules/buckets"
+  jobs_bucket        = "${data.external.secrets.result["jobs_bucket"]}"
+  ais_bucket         = "${data.external.secrets.result["ais_bucket"]}"
+  heatmap_bucket     = "${data.external.secrets.result["heatmap_bucket"]}"
+  emr_logs_bucket    = "${data.external.secrets.result["emr_logs_bucket"]}"
+  spark_job_jar      = "${var.SPARK_JOB_JAR}"
+  spark_job_jar_name = "${var.SPARK_JOB_JAR_NAME}"
```
Comment (andyphelps) [inline deployment/src/main/terraform/main.tf]: What is the difference betwwen spark_job_jar and spark_job_jar_name?  It's not obvious

File: deployment/build.gradle.kts (hunk 1 above)
Comment (JoelClemence) [reply deployment/build.gradle.kts]: done

File: deployment/src/main/terraform/main.tf (hunk 2 above)
Comment (JoelClemence) [reply deployment/src/main/terraform/main.tf]: Renamed other variable making it clear the other is the full path to jar

**Pull request merged**